components/forum/DiscourseAdapter.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+namespace app\components\forum;
+
+use app\models\User;
+use Yii;
+use yii\base\Component;
+use yii\base\InvalidConfigException;
+use yii\db\Connection;
+use yii\db\Query;
+use yii\di\Instance;
+use yii\helpers\ArrayHelper;
+use yii\helpers\Inflector;
+use yii\httpclient\Client;
+
+/**
+ * DiscourseAdapter implements a forum bridge between the Discourse Forum and the application.
+ * Configure as follows:
+ *
+ * ```php
+ * 'forumBridge' => [
+ *      'class' => \app\components\forum\DiscourseAdapter::class,
+ *      'apiUrl' => 'https://forum.yiiframework.com/',
+ *      'apiToken' => '123456',
+ *  ],
+ * ```
+ *
+ * @see https://docs.discourse.org/
+ */
+class DiscourseAdapter extends Component implements ForumAdapterInterface
+{
+    /**
+     * @var string discourse API URL.
+     */
+    public $apiUrl;
+    /**
+     * @var string discourse API auth token.
+     */
+    public $apiToken;
+    /**
+     * @var string discourse API user name for requests that need admin permission.
+     */
+    public $apiAdminUser = 'cebe';
```

Comment:
Hardcode?

Reply:
that's the default, you can override it in your config :)